### ISSUE TITLE
Potential Unfairness in Testing Procedure

### ISSUE THREAD

**Issue by Mango123L** [cong-zeng/ood-llm-detect#10]
Hi, thank you for this excellent work! I've been studying your code and found it very well-structured. However, I noticed one issue in `test.py` that may affect the reliability and fairness of the reported results:

**The code optimizes the classification threshold on the test set itself:**

```python
threshold, f1 = best_threshold_by_f1(label_np, pred_np)
y_pred = np.where(pred_np>threshold,1,0)
```

I'm concerned this **might introduce unfairness in comparison with baseline methods**. Typically, other detection methods either use thresholds determined from validation sets or report threshold-independent metrics like AUC-ROC and AUC-PR. By optimizing the threshold directly on the test set, this approach effectively **performs hyperparameter tuning on test data**, which could give an advantage that other methods don't have. 

Could you help clarify how the baseline methods in your comparison handle thresholds? If they use validation-based or fixed thresholds, would it be more appropriate to adopt a similar approach for fair comparison?

---

***This issue may impact the credibility of reported results and fair comparison with baselines.*** Looking forward to your clarification on these concerns. Thank you!
